ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiCommandService.scala

```diff
@@ -154,11 +124,13 @@ private[apiserver] final class ApiCommandService private[services] (
   )(implicit
       loggingContext: LoggingContext,
       errorLogger: ContextualizedErrorLogger,
-  ): Future[Either[TrackedCompletionFailure, CompletionSuccess]] =
+  ): Future[CompletionResponse] =
     if (running) {
       val timeout = Option(Context.current().getDeadline)
         .map(deadline => Duration.ofNanos(deadline.timeRemaining(TimeUnit.NANOSECONDS)))
-      submissionTracker.track(CommandSubmission(request.getCommands, timeout))
+        .map(ApiCommandService.durationOrdering.max(_, maximumTrackingTimeout))
+        .getOrElse(maximumTrackingTimeout)
+      submissionTracker.track(request.commands.get, timeout, submit) // TODO is .get ok?
```

Comment:
theoretically, yes, since we have the validation upstream, no? Maybe folding with a specialized error would be better suited